src/main/java/uk/gov/hmcts/reform/sscs/service/SaveLetterCorrespondenceAsyncService.java

```diff
@@ -0,0 +1,46 @@
+package uk.gov.hmcts.reform.sscs.service;
+
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.retry.annotation.Backoff;
+import org.springframework.retry.annotation.Recover;
+import org.springframework.retry.annotation.Retryable;
+import org.springframework.scheduling.annotation.Async;
+import org.springframework.stereotype.Component;
+import uk.gov.hmcts.reform.sscs.ccd.domain.Correspondence;
+import uk.gov.service.notify.NotificationClient;
+import uk.gov.service.notify.NotificationClientException;
+
+@Slf4j
+@Component
+public class SaveLetterCorrespondenceAsyncService {
+    private final CcdNotificationsPdfService ccdNotificationsPdfService;
+
+
+    @Autowired
+    public SaveLetterCorrespondenceAsyncService(CcdNotificationsPdfService ccdNotificationsPdfService) {
+        this.ccdNotificationsPdfService = ccdNotificationsPdfService;
+    }
+
+    @Async
+    @Retryable(maxAttemptsExpression =  "#{${letterAsync.maxAttempts}}", backoff = @Backoff(delayExpression = "#{${letterAsync.delay}}", multiplierExpression = "#{${letterAsync.multiplier}}", random = true))
```

Comment:
`random = true` seems to fix an issue I had locally.

> ccd-data-store-api_1              | 2019-10-02T14:06:37.157 INFO  [http-nio-4452-exec-8] u.g.h.c.d.casedetails.DefaultCaseDetailsRepositoryjavax.persistence.OptimisticLockException: Row was updated or deleted by another transaction (or unsaved-value mapping was incorrect) : [uk.gov.hmcts.ccd.data.casedetails.CaseDetailsEntity#127]

I could no longer reproduce this when this was set. I'd much rather to this than add a synchronised block.